core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1106,6 +1159,26 @@ else if (isInferenceCandidate(conjunct)) {
                     TRUE);
         }
 
+        private boolean isSimpleExpression(Expression expression)
```

Comment:
static ?

Reply:
I think you can move this to `IrUtils` class

Reply:
Because it is vague, I'm not sure this is useful as a general functionality.